### PR TITLE
PIK-4274 Remove url path prefix from signature payload

### PR DESCRIPTION
**Pull request by shakedfb**

### CODE REVIEW

File: v2/api-validator/src/client/SecureClient.ts [hunk 1]
```diff
@@ -228,7 +228,9 @@ export function createSecurityHeaders(
 
 function getRelativeUrl(url: string) {
   const parsedUrl = new URL(url);
-  return parsedUrl.pathname + parsedUrl.search;
+  const relativeUrl = parsedUrl.pathname + parsedUrl.search;
+  const prefix = config.getServerUrlPrefix();
+  return relativeUrl.replace(prefix, '');
```
Comment (amirhac) [inline v2/api-validator/src/client/SecureClient.ts:233]: I prefer using `relativeUrl.slice(prefix.length)`

Comment (amirhac) [reply v2/api-validator/src/client/SecureClient.ts:233]: Consider move it to some `path-util` file

File: v2/api-validator/src/config/index.ts [hunk 2]
```diff
@@ -244,4 +244,9 @@ config.validate({ allowed: 'strict' });
 config.getUnifiedOpenApiPathname = () =>
   path.join(config.get('openApi').location, config.get('openApi').unifiedFilename);
 
+config.getServerUrlPrefix = () => {
+  const prefix = new URL(config.get('client.serverBaseUrl')).pathname;
+  return prefix.endsWith('/') ? prefix.slice(0, -1) : prefix;
```
Comment (amirhac) [inline v2/api-validator/src/config/index.ts:249]: Consider moving it to some `path-util` file

File: v2/api-validator/src/server/middlewares/verify-signature-middleware.ts [hunk 3]
```diff
@@ -42,6 +43,11 @@ function getSignatureHeaders(headers: IncomingHttpHeaders) {
   return { timestamp, nonce, signature };
 }
 
+function removeUrlPrefix(url: string) {
+  const prefix = config.getServerUrlPrefix();
+  return url.replace(prefix, '');
```
Comment (amirhac) [inline v2/api-validator/src/server/middlewares/verify-signature-middleware.ts:48]: Should be moved to a util file

**Review by amirhac — CHANGES_REQUESTED** [review]

File: v2/api-validator/src/client/SecureClient.ts (hunk 1 above)
Comment (shakedfb) [reply v2/api-validator/src/client/SecureClient.ts]: Thanks, move to a different file

**Review by amirhac — APPROVED** [review]

**Pull request merged**